llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp

```diff
@@ -7980,6 +7980,63 @@ static Instruction *foldFabsWithFcmpZero(FCmpInst &I, InstCombinerImpl &IC) {
   }
 }
 
+/// Optimize sqrt(X) compared with zero.
+static Instruction *foldSqrtWithFcmpZero(FCmpInst &I, InstCombinerImpl &IC) {
+  Value *X;
+  if (!match(I.getOperand(0), m_Sqrt(m_Value(X))))
+    return nullptr;
+
+  if (!match(I.getOperand(1), m_PosZeroFP()))
```

Comment:
This should work for anyZeroFP

Reply:
Most: https://alive2.llvm.org/ce/z/nwWqUB

Reply:
fcmp with -0.0 is not canonical:

https://github.com/llvm/llvm-project/blob/10df3207434e603be5f7e9b3036d821dd5623d3a/llvm/lib/Transforms/InstCombine/InstCombineCompares.cpp#L8135-L8138